src/pages/user_profile/AffiliationAndRoles.jsx

```diff
@@ -1,190 +1,39 @@
 import React from 'react';
 import { useEffect, useState } from 'react';
 import { Institution } from '../../libs/ajax/Institution';
-import { User } from '../../libs/ajax/User';
-import { find, isNil, isNumber } from 'lodash';
-import { Notifications, setUserRoleStatuses } from '../../libs/utils';
-import { Storage } from '../../libs/storage';
-import { FormField, FormFieldTypes } from '../../components/forms/forms';
-import Tooltip from '@mui/material/Tooltip';
-import { ThemeProvider, createTheme } from '@mui/material/styles';
+import { Notifications } from '../../libs/utils';
 
 export default function AffiliationAndRole(props) {
 
-  const theme = createTheme({
-    components: {
-      MuiTooltip: {
-        styleOverrides: {
-          tooltip: {
-            fontSize: '14px',
-            color: 'white',
-            fontFamily: 'Montserrat'
-          }
-        }
-      },
-      MuiButton: {
-        root: {
-          fontSize: '14px',
-          color: 'white',
-          fontFamily: 'Montserrat',
-          background: 'blue',
-          '&:hover': {
-            backgroundColor: 'darkblue',
-          },
-        }
-      }
-    }
-  });
-
-  const {
-    user,
-    userProps,
-    institutions
-  } = props;
-
-  const [profile, setProfile] = useState({
-    roles: '',
-    institutionId: undefined,
-    suggestedInstitution: undefined,
-    selectedSigningOfficialId: undefined,
-    suggestedSigningOfficial: undefined,
-    id: undefined
-  });
-
-  const [selectedInstitution, setSelectedInstitution] = useState();
+  const { user } = props;
+  const [institution, setInstitution] = useState(undefined);
+  const [roles, setRoles] = useState([]);
 
   useEffect(() => {
     const init = async () => {
       try {
-        const rolesList = [];
-        if (!isNil(user) && !isNil(user.roles)) {
-          for (let i = 0; i < user.roles.length; i++) {
-            const newRole = user.roles[i].name;
-            rolesList.push(newRole);
+        const allRoles = user?.roles?.map((role) => role.name).join(', ');
```

Comment:
This is good but it would be nice to typescriptfiy this too when we have time, otherwise it's hard to know if the `?`s are required or not

Reply:
I'll give that a shot ... moving components jsx to tsx have typically been non-trivial in this codebase